### PR TITLE
Bug Fix - crop needs to be set using target width and height

### PR DESCRIPTION
**Pull request by LakshmiKumar23**
Fixes a seg fault observed when using crop parameters for video decode reported by @paveltc. Tested with cropping middle of the image, start and towards the end. Requires this change since we now have `disp_height/disp_width` and `target_height/target_width` separate

### CODE REVIEW

**Comment by jeffqjiangNew** [comment]
@LakshmiKumar23 Do AVC and HEVC conformance tests pass with this PR?

**Comment by LakshmiKumar23** [comment]
@jeffqjiangNew I did not run conformance. Did you mean to close this PR?

**Comment by jeffqjiangNew** [comment]
@LakshmiKumar23 A good practice before pushing changes to rcoDecode is to run AVC and HEVC conformance tests and make sure they pass. Particularly in this PR, we modify output surface info and chroma height, which may (or may not) impact how we output decoded images when user cropping is not set.

File: utils/rocvideodecode/roc_video_dec.cpp [hunk 1]
```diff
@@ -377,7 +377,7 @@ int RocVideoDecoder::HandleVideoSequence(RocdecVideoFormat *p_video_format) {
     videoDecodeCreateInfo.target_width = target_width_;
     videoDecodeCreateInfo.target_height = target_height_;
 
-    chroma_height_ = (int)(ceil(disp_height_ * GetChromaHeightFactor(video_surface_format_)));
+    chroma_height_ = (int)(ceil(target_height_ * GetChromaHeightFactor(video_surface_format_)));
```
Comment (jeffqjiangNew) [inline utils/rocvideodecode/roc_video_dec.cpp:380]: chroma_height_ is also set in re-configure call. If we change its calculation here, we need to change in re-configure too.

chroma_height_ is loosely defined and used in quite a few places. This is a good opportunity to review its define and usage in call cases.

**Review by jeffqjiangNew — CHANGES_REQUESTED** [review]

File: utils/rocvideodecode/roc_video_dec.cpp (hunk 1 above)
Comment (rrawther) [reply utils/rocvideodecode/roc_video_dec.cpp:380]: Yes, please check all the places where output dimensions are calculated and set with and without crop

**Review by rrawther — COMMENTED** [review]
added a comment

**Comment by LakshmiKumar23** [comment]
@jeffqjiangNew @rrawther ran conformance with this branch. All 135 streams are passing
```
Conformance test completed on the 135 streams:
     - The number of passing streams is 135
     - The number of failing streams is 0
     - The number of streams that did not finish decoding is 0

```

File: utils/rocvideodecode/roc_video_dec.cpp [hunk 2]
```diff
@@ -387,8 +387,8 @@ int RocVideoDecoder::HandleVideoSequence(RocdecVideoFormat *p_video_format) {
     }
     chroma_vstride_ = (int)(ceil(surface_vstride_ * GetChromaHeightFactor(video_surface_format_)));
     // fill output_surface_info_
-    output_surface_info_.output_width = disp_width_;
-    output_surface_info_.output_height = disp_height_;
+    output_surface_info_.output_width = target_width_;
+    output_surface_info_.output_height = target_height_;
```
Comment (jeffqjiangNew) [inline utils/rocvideodecode/roc_video_dec.cpp:391]: Probably need to do the same in re-configure call.

**Review by jeffqjiangNew — CHANGES_REQUESTED** [review]

**Comment by jeffqjiangNew** [comment]
Can you also test AVC conformance streams as well?

**Comment by LakshmiKumar23** [comment]
@jeffqjiangNew you closed the PR again haha

**Comment by jeffqjiangNew** [comment]
@LakshmiKumar23 Sorry I probably clicked the wrong button. My bad. :)

**Comment by LakshmiKumar23** [comment]
> Can you also test AVC conformance streams as well?

AVC passes too, all 127 streams
```


Conformance test completed on the 127 streams:
     - The number of passing streams is 127
     - The number of failing streams is 0
     - The number of streams that did not finish decoding is 0

```

**Comment by LakshmiKumar23** [comment]
@paveltc can you run some other tests on this PR too?

**Review by rrawther — APPROVED** [review]

**Review by jeffqjiangNew — APPROVED** [review]

**Pull request merged**